Update installation.md to fix a confusion that whether synapse can serve Client Well-Known URI (#16678)
matrix-org/synapse: docs/setup/installation.md

`Access-Control-Allow-Origin: *` which would allow all browser based clients to
view it.

> Synapse will serve it once you set `public_baseurl` in your `homeserver.yaml`.
> 
> Just like:
> 
> ```yaml
> public_baseurl: "https://<matrix.example.com>"
> ```
> 
> Alternatively, you can set your reverse proxy to serve it.

Reviewer: ```suggestion
Synapse will generate a response if `public_baseurl` is configured in `homeserver.yaml`:

```yaml
public_baseurl: "https://<matrix.example.com>"
```

Alternatively, you can set your reverse proxy to serve it.
```

Reviewer: Should this cross-reference the configuration manual w/ the various settings it can include?

}
```

You should always ensure the `public_baseurl` option in `homeserver.yaml` is set
correctly. `public_baseurl` should be set to the URL that clients will use to
connect to your server. If you use reverse proxy to serve client Well-Known URI,
you should ensure the URL you put for the `public_baseurl` in `homeserver.yaml`

> and `m.homeserver` `base_url` in your reverse proxy config must be the same.

Reviewer: I find this paragraph a bit confusing now... it *might* be clearer to do the following:

1. Add a new sub-heading (under "Client Well-Known URI") which is "Serve client well-known with Synapse", this would contain the new bits you added about this.
2. Put the reverse proxy bits into a new sub-heading "Serve client well-known with reverse proxy"

Maybe include the `public_baseurl` before either section then since it is needed for both?
